topi/python/topi/nn/softmax.py

```diff
@@ -4,20 +4,27 @@
 import tvm
 
 @tvm.tag_scope(tag='softmax_output')
-def softmax(x):
+def softmax(x, axis=-1):
     """Perform softmax activation on the data
```

Comment:
I think it is likely to get softmax to work for simple axis argument without enumerate the axis parameter.

Take a look at the implementation of reduction (with axis parameter)

Reply:
well, softmax itself doesn't need axis parameter, it it knows the input layout.

here, I need axis param to switch between NCHW or NHWC softmax.

Reply:
What I mean is that the axis parameter is better than the layout(NCHW or NHWC) parameter, since it is more general. axis=-1 means NCHW, axis=3  means NHWC

We only need to improve the original implementation to properly support more than 2D case

Reply:
ok, will clean up softmax implementation